### PR TITLE
3020 typography impact eval

### PR DESCRIPTION
**Pull request by halocline**
#### What does this PR do?

Merging into NEXT branch for impact / evaluation. Applies modified typography sizes, line-heights, weights, and colors for headings, paragraphs, text, and form field inputs.

To investigate:
- [ ] buttons
- [ ] anchors
- [ ] accordion
- [ ] menu
- [ ] tabs
- [ ] tip
- [ ] data table
- [ ] name-value list
- [ ] tag

Questions:
- Do text-weak and text-xweak values need to be adjusted to play better with new text-strong and text values?
- FormField text styling - probably should bump weight down for label and input value. Input value color and placeholder color are too similar when at lighter weight. Use new text-strong for input value color?
- Do border colors need to be adjusted to play better with thinner font weights?
- Current border colors were recently made with opacity. This causes issues when borders overlap such as in Accordion where adjacent panel borders appear darker than edge panels.
- Re-calculate maxWidths for paragraph sizes?
- Chris is introducing a new type of typography “quote (e.g. positioning statement),” is this a typography type we want to add to Grommet?
- Not touching button text for now…
- Icon colors

#### What testing has been done on this PR?

Implemented within aries.js first and visually tested on the Design System site.

#### Any background context you want to provide?

https://www.figma.com/file/Ceilu1kjl81OE8FMwz12op/HPE-Unified-Campaign?node-id=1056%3A20774&t=8wAZQCPMnQSABCts-0

#### What are the relevant issues?

https://github.com/grommet/hpe-design-system/issues/3020

#### Is this change backward compatible or could it be a breaking change for the official HPE theme?

Probably a breaking change as there will likely be noticeable effects on content layouts do to changing line heights which may require projects to revisit gap in some areas.

### CODE REVIEW

File: src/js/themes/hpe.js [hunk 1]
```diff
@@ -81,7 +81,7 @@ export const hpe = deepFreeze({
     input: {
       font: {
         height: 'inherit',
-        weight: 500,
+        weight: 450, // weight: 500,
```
Comment (ericsoderberghp) [inline src/js/themes/hpe.js:84]: I didn't see 450 in the list of font weights we're asking for in the CDN.

File: src/js/themes/hpe.js [hunk 2]
```diff
@@ -103,17 +103,37 @@ export const hpe = deepFreeze({
     },
     font: {
       family: "'Metric', Arial, sans-serif",
+      // This face configuration is intended for typography exploration. Proper config
+      // will need enhancements to Grommet.
       face: `
+        @font-face {
+          font-family: "Metric";
+          src: url("https://d3hq6blov2iije.cloudfront.net/fonts/HPEXS-Metric-Fonts/MetricHPEXSLight-Regular.woff2") format('woff2'),
```
Comment (ericsoderberghp) [inline src/js/themes/hpe.js:111]: these seem to be duplicating the URLs for the `300` weight.

File: src/js/themes/hpe.js [hunk 3]
```diff
@@ -874,141 +894,160 @@ export const hpe = deepFreeze({
...
   heading: {
     color: 'text-strong',
+    weight: 450,
     level: {
       1: {
         small: {
-          size: '36px',
-          height: '42px',
+          size: '48px', // size: '36px',
+          height: '48px', // height: '42px',
```
Comment (ericsoderberghp) [inline src/js/themes/hpe.js:902]: I'm wondering if we should adjust the theme to allow setting `heading.height: '1em'` generally, which would then apply to all levels and weights.

File: src/js/themes/hpe.js [hunk 4]
```diff
@@ -874,141 +894,160 @@ export const hpe = deepFreeze({
...
           maxWidth: '854px',
         },
         medium: {
-          size: '53px',
-          height: '59px',
+          size: '72px', // size: '53px',
+          height: '72px', // height: '59px',
```
Comment (ericsoderberghp) [inline src/js/themes/hpe.js:907]: Given that we're setting line height to 1em, we should test with heading text that wraps, to ensure that the multiple lines don't appear too squished.

File: src/js/themes/hpe.js [hunk 5]
```diff
@@ -1448,33 +1490,37 @@ export const hpe = deepFreeze({
   },
   text: {
     xsmall: {
+      // weight needs to be modified at the size level
       size: '14px',
-      height: '20px',
-      maxWidth: '326px',
+      height: '18px',
+      maxWidth: '379px',
```
Comment (ericsoderberghp) [inline src/js/themes/hpe.js:1496]: Why are we boosting maxWidth to align with the "small" case? I was expecting the maxWidth to be generally proportional to the font size.

**Review by ericsoderberghp — COMMENTED** [review]
nice start, some initial comments and questions

File: src/js/themes/hpe.js (hunk 1 above)
Comment (taysea) [reply src/js/themes/hpe.js:84]: I believe this is a temp fix related to how the font-family and weight definitions needed to be done to allow 300 to be default weight. Matt mentioned he might have to fudge some things on this PR but that any grommet enhancements would be done once the typography direction is nailed dow.

File: src/js/themes/hpe.js (hunk 2 above)
Comment (taysea) [reply src/js/themes/hpe.js:111]: ^^ I believe this is also related to: https://github.com/grommet/grommet-theme-hpe/pull/291#discussion_r1026642955

File: src/js/themes/hpe.js (hunk 1 above)
Comment (halocline) [reply src/js/themes/hpe.js:84]: Yes, this is a temporary hack for preview purposes only.

The typography direction calls for Metric-light to be the new default. However, Grommet Text does not currently allow weight to be defined by the theme (only family).

For expediency, I'm proposing we use this configuration for assessing the typography styling impacts and cataloging the needed work to support.

File: src/js/themes/hpe.js (hunk 2 above)
Comment (halocline) [reply src/js/themes/hpe.js:111]: Same as the previous comment:

Yes, this is a temporary hack for preview purposes only.

The typography direction calls for Metric-light to be the new default. However, Grommet Text does not currently allow weight to be defined by the theme (only family).

For expediency, I'm proposing we use this configuration for assessing the typography styling impacts and cataloging the needed work to support.

File: src/js/themes/hpe.js (hunk 4 above)
Comment (halocline) [reply src/js/themes/hpe.js:907]: I'm cataloging a list of questions and components to evaluate in the main issue: https://github.com/grommet/hpe-design-system/issues/3020#issue-1448456130

File: src/js/themes/hpe.js (hunk 5 above)
Comment (halocline) [reply src/js/themes/hpe.js:1496]: That's a copy-paste error on my side. I will fix.

With the shift to thinner font weights, I think we should re-evaluate the maxWidths throughout to ensure they fit to the readability line length recommendations. Item in the [todo list](https://github.com/grommet/hpe-design-system/issues/3020#issue-1448456130).

**Review by taysea — APPROVED** [review]

**Pull request merged**